Gemfile.lock

```diff
@@ -19,21 +33,74 @@ GEM
       minitest (>= 5.1)
       tzinfo (~> 2.0)
       zeitwerk (~> 2.3)
+    addressable (2.7.0)
+      public_suffix (>= 2.0.2, < 5.0)
+    aws-eventstream (1.1.0)
+    aws-partitions (1.422.0)
+    aws-sdk-core (3.112.0)
+      aws-eventstream (~> 1, >= 1.0.2)
+      aws-partitions (~> 1, >= 1.239.0)
+      aws-sigv4 (~> 1.1)
+      jmespath (~> 1.0)
+    aws-sdk-glacier (1.36.0)
```

Comment:
@Azdaroth Ideally it shouldn't be included in Gemfile as long as you don't use it.
Imagine there are 10 of different options where you can store archived data. Each of them uses it's own lib to push archived data. But you use only a subset of storages.
On my mind it should load only those gems which are used actually.
As a possible solution we may add new gems for each new remote storage:

```ruby
# in Gemfile
gem 'tartarus-rb'
gem 'tartarus-rb-aws-glacier-remote-storage'
gem 'tartarus-rb-my-super-remote-storage'
```

This way if we add new remote storage class but don't add gem we'd get an unitialized constant error.

Reply:
@sineed I definitely agree with that, it's just the fact it was better to have it done this way and have the Glacier support faster than to not have it done and wait even more. If you have some spare time to submit a PR to make it better, feel free to handle it, I would be definitely happy to merge it and release a new version. Probably the most optimal solution here would be to not make aws-sdk-glacier a dependency, and force consumer of the gem to use require it on their own by raising error whenever a Glacier storage class is initialized without this gem.